photon-server/src/main/java/org/photonvision/vision/opencv/CVMat.java

```diff
@@ -43,23 +44,34 @@ public void copyTo(Mat srcMat) {
         srcMat.copyTo(mat);
     }
 
+    public void logStackTrace() {
+        var trace = Thread.currentThread().getStackTrace();
+
+        final var traceStr = new StringBuilder();
+        for (int idx = 3; idx < trace.length; idx++) {
+            traceStr.append("\t\n").append(trace[idx]);
+        }
+        logger.trace("\n" + traceStr.toString() + "\n");
+    }
+
     public CVMat(Mat mat) {
         this.mat = mat;
-        if (allMats.add(mat) && shouldPrint) {
-            var trace = Thread.currentThread().getStackTrace();
-
-            final var traceStr = new StringBuilder();
-            for (var elem : trace) {
-                traceStr.append("\t").append(elem);
-            }
-            logger.trace(traceStr::toString);
+        allMatCounter++;
+        allMats.put(mat, allMatCounter);
+
+        if (shouldPrint) {
+            logger.trace(() -> "CVMat" + allMatCounter + " alloc - new count: " + allMats.size());
+            logStackTrace();
         }
     }
 
     @Override
     public void release() {
+        int matNo = allMats.get(mat);
         allMats.remove(mat);
         mat.release();
+        logger.trace(() -> "CVMat" + matNo + " de-alloc - new count: " + allMats.size());
```

Comment:
Should this be wrapped in a check for `shouldPrint`?

Reply:
likely

Reply:
yes and no - it won't print unless the Logger has been set to TRACE

Reply:
Per conversation - fixed it up to use lambdas better for performance.